x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/autodetect/params/AutodetectParams.java

```diff
@@ -124,6 +135,11 @@ public Builder setModelSizeStats(ModelSizeStats modelSizeStats) {
             return this;
         }
 
+        public Builder setTimingStats(TimingStats timingStats) {
+            this.timingStats = timingStats;
```

Comment:
```suggestion
            this.timingStats = new TimingStats(timingStats);
```

Reply:
@przemekwitek `TimingStats` has a method that allows its internal state to mutate. Not doing a copy would allow the parameter `timingStats#updateStats` to be called AFTER it is passed to this method causing the `Builder#timingStats` to be updated as well. 

https://repl.it/@ben_w_trent/mutateState

Reply:
I understand that. It's just this setter in builder is only called once and the temporary object is not saved anywhere else:
paramsBuilder.setTimingStats(parseSearchHit(hit, TimingStats.PARSER, errorHandler));

But ok, I applied your suggestion to prevent some random refactoring of breaking this assumption.